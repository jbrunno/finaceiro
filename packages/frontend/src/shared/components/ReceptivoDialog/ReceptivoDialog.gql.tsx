import { gql } from '@apollo/client';

export const CLIENTE_INSERT_MUTATION = gql`
  mutation ClientePost($input: ClientePostInput!) {
    clientePost(input: $input) {
      id
    }
  }
`;

export type ClienteInsertInfoInput = {
  nome: string;
  telefone: string;
  tipoCliente: string;
  documento: string | null;
};
export type ClienteInsertInfoData = {
  clientePost: {
    id: string;
  };
};
