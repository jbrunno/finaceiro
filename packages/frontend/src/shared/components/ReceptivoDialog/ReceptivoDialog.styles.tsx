import { styled, css } from '@frontend/styles';
import {
  Button,
  DialogContent,
  InternationalPhoneField,
  RadioGroup,
  TextField,
} from '@frontend/components';

export const Header = styled('div')(
  ({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    gap: ${theme.fn.spacing(8)};
    max-width: ${theme.fn.size(217)};
    margin: 0 auto;
    margin-bottom: ${theme.fn.spacing(16)};
  `,
);

export const TextFieldStyled = styled(TextField)`
  width: 100%;
  min-height: ${({ theme }) => theme.fn.size(60)};
  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
`;

export const StyledForm = styled('form')(
  ({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.fn.spacing(16)};
    padding: ${theme.fn.spacing(8, 16)};
    min-width: ${theme.fn.size(300)};
  `,
);

export const RadioGroupStyled = styled(RadioGroup)(
  ({ theme }) => css`
    display: flex;
    flex-direction: row;
    gap: ${theme.fn.spacing(8)};
    margin-bottom: ${theme.fn.spacing(12)};
  `,
);

export const ButtonStyled = styled(Button)`
  width: 100%;
  margin: ${({ theme }) => theme.fn.spacing(0, 32, 24)};
`;

export const InternationalPhoneFieldStyled = styled(InternationalPhoneField)`
  min-height: ${({ theme }) => theme.fn.size(65)};
`;

export const DialogContentStyled = styled(DialogContent)`
  padding-bottom: 0;
`;
