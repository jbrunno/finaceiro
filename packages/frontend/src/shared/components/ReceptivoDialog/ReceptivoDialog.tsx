import {
  Typography,
  Dialog,
  DialogActions,
  DialogProps,
  FormControlLabel,
  Radio,
} from '@frontend/components';
import { ClienteTipoEnum } from '@/shared/constants';
import {
  ButtonStyled,
  Header,
  TextFieldStyled,
  RadioGroupStyled,
  StyledForm,
  InternationalPhoneFieldStyled,
  DialogContentStyled,
} from './ReceptivoDialog.styles';
import {
  FormFields,
  UseReceptivoDialogProps,
  useReceptivoDialog,
} from './hooks/useReceptivoDialog';

type ReceptivoDialogProps = {
  className?: string;
} & Omit<DialogProps, 'children'> &
  UseReceptivoDialogProps;

export function ReceptivoDialog({
  onClose,
  onSaved,
  ...dialogProps
}: ReceptivoDialogProps) {
  const { formik, typeInput, handleChangeDocumento, maxLength } =
    useReceptivoDialog({
      onSaved,
      onClose,
    });

  return (
    <Dialog {...dialogProps} onClose={onClose} maxWidth="xl">
      <DialogContentStyled>
        <Header>
          <Typography variant="h3" color="textPrimary">
            Atendimento Receptivo Iniciado
          </Typography>
          <Typography variant="body1" color="textSecondary">
            NÃ£o encontramos um cliente associado a este telefone.
          </Typography>
        </Header>

        <StyledForm onSubmit={formik.handleSubmit}>
          <InternationalPhoneFieldStyled
            size="small"
            color="secondary"
            country="br"
            specialLabel="Telefone"
            value={formik.values.telefone}
            onBlur={formik.handleBlur(FormFields.telefone)}
            onChange={(value) =>
              formik.setFieldValue(FormFields.telefone, value)
            }
            error={(formik.touched.telefone && formik.errors.telefone) || ''}
          />
          <TextFieldStyled
            variant="outlined"
            label="Nome completo do cliente"
            size="small"
            value={formik.values.nome}
            onBlur={formik.handleBlur(FormFields.nome)}
            onChange={formik.handleChange(FormFields.nome)}
            error={!!(formik.touched.nome && formik.errors.nome)}
            errorText={formik.errors[FormFields.nome]}
          />
          <RadioGroupStyled
            name={FormFields.tipo}
            value={String(formik.values.tipo)}
            onChange={(e) => {
              formik.setFieldValue(FormFields.documento, '');
              formik.setFieldValue(FormFields.tipo, e.target.value);
            }}
          >
            <FormControlLabel
              value={ClienteTipoEnum.FISICA}
              control={<Radio color="primary" />}
              label="CPF"
            />
            <FormControlLabel
              value={ClienteTipoEnum.JURIDICA}
              control={<Radio color="primary" />}
              label="CNPJ"
            />
            <FormControlLabel
              value={ClienteTipoEnum.ESTRANGEIRA}
              control={<Radio color="primary" />}
              label="ID/Passaporte"
            />
          </RadioGroupStyled>
          <TextFieldStyled
            variant="outlined"
            label="CPF/CNPJ/ID ou Passaporte"
            size="small"
            value={formik.values.documento}
            onBlur={formik.handleBlur(FormFields.documento)}
            onChange={(value) => handleChangeDocumento(value.target.value)}
            type={typeInput}
            error={!!formik.errors.documento}
            errorText={formik.errors[FormFields.documento]}
            inputProps={{ maxLength }}
          />
        </StyledForm>
      </DialogContentStyled>
      <DialogActions>
        <ButtonStyled
          onClick={formik.submitForm}
          size="large"
          variant="contained"
          color="primary"
          disabled={!formik.dirty || (formik.dirty && !formik.isValid)}
        >
          Cadastrar cliente
        </ButtonStyled>
      </DialogActions>
    </Dialog>
  );
}
