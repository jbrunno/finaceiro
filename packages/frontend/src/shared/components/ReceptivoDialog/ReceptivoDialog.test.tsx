import { render } from '@test-utils/component-testing';
import { ReceptivoDialog } from './ReceptivoDialog';

test('should render ReceptivoDialog', () => {
  const { container } = render(<ReceptivoDialog open onClose={() => {}} />);
  expect(container).toMatchSnapshot();
});
