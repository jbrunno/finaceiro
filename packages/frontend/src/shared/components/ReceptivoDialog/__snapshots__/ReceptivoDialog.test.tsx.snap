// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ReceptivoDialog 1`] = `
<div
  aria-hidden="true"
/>
`;
